core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/TitleBlockServices.java

```diff
@@ -804,48 +807,89 @@ public String getTitleBockCellLabel(DAnnotation cell) {
     return cell.getDetails().get(TitleBlockHelper.CONTENT);
   }
 
-  /**
-   * show or hide a Title Block depending on given type (DIAGRAM_TITLE_BLOCK or ELEMENT_TITLE_BLOCK). selectedTypes are
-   * the elements that shall be displayed
-   * 
-   * @param elementView
-   * @param selectedTitleBlocks
-   * @param type
-   * @return String the label of the content cell
-   */
-  public EObject showHideTitleBlocks(final EObject elementView, Collection<DAnnotation> selectedTitleBlocks) {
-    DDiagram diagram = CapellaServices.getService().getDiagramContainer(elementView);
-    Map<DAnnotation, DDiagramElement> existingTitleBlocks = new HashMap<>();
-    if (elementView instanceof EdgeTarget) {
-      // we want to show/hide the Title Block associated to elementView
-      List<DAnnotation> elementTitleBlocks = getAssociatedElementTitleBlocks((EdgeTarget) elementView);
-      elementTitleBlocks.stream().forEach(
-          tb -> existingTitleBlocks.put(tb, DiagramServices.getDiagramServices().getDiagramElement(diagram, tb)));
+  public EObject showHideTitleBlocks(EObject targetView, Collection<DAnnotation> titleBlockUserSelection) {
+
+    DDiagram diagram = CapellaServices.getService().getDiagramContainer(targetView);
+    Map<DAnnotation, Collection<DSemanticDecorator>> existingTitleBlocksToViews = Collections.emptyMap();

```

Comment:
We now extract all the DSemanticDecorators that have Title Blocks as target (Edges and Containers), so we properly hide everything.